p2p/peermanager.go

```diff
@@ -400,17 +401,32 @@ func (m *PeerManager) Add(address NodeAddress) error {
 	if !ok {
 		peer = m.newPeerInfo(address.NodeID)
 	}
-	if _, ok := peer.AddressInfo[address]; !ok {
-		peer.AddressInfo[address] = &peerAddressInfo{Address: address}
+	_, ok = peer.AddressInfo[address]
+	// if we already have the peer address, there's no need to continue
+	if ok {
+		return false, nil
 	}
+
+	// else add the new address
+	peer.AddressInfo[address] = &peerAddressInfo{Address: address}
 	if err := m.store.Set(peer); err != nil {
-		return err
+		return false, err
 	}
 	if err := m.prunePeers(); err != nil {
-		return err
+		return true, err
 	}
 	m.dialWaker.Wake()
-	return nil
+	return true, nil
+}
+
+// Capacity returns the ratio of peer addresses stored to the maximum size.
+func (m *PeerManager) Capacity() float64 {
```

Comment:
I'm not sure `Capacity` is the best API here. Maybe `ConnectedPeerRatio`?

Reply:
But it's not just connected peers. It's the ratio of all peers to to the max amount of peers

Reply:
Yeah sure, but capacity doesn't really allude to that. Maybe just `PeerRatio`?

Reply:
IMO, I think it does. If we set a max of 100 peers, and we currently have 50 saved then we're at half of the capacity and this function will return 0.5.

But I'm not really fussed about this either so I can rename it to `PeerRatio`